core/state/snapshot/conversion.go

```diff
@@ -321,7 +321,7 @@ func generateTrieRoot(db ethdb.KeyValueWriter, it Iterator, account common.Hash,
 						results <- err
 						return
 					}
-					if !bytes.Equal(account.Root, subroot.Bytes()) {
+					if checkSubRoot && !bytes.Equal(account.Root, subroot.Bytes()) {
```

Comment:
Aren't the subtries calculated in `leafCallback` ? If so, aren't storage tries computed no matter the value of `checkSubRoot` ?

Reply:
So this condition makes sure the computed storage root equals the one stored in `account.storageRoot`. But in our case this never happens because verkle roots are different.